EpiAware/src/EpiInfModels/ODEProcess.jl

```diff
@@ -0,0 +1,170 @@
+the incubation rate `α` and the recovery rate `γ`.
+
+```julia
+using EpiAware, OrdinaryDiffEq, Distributions
+# Define the time span for the ODE problem
+tspan = (0.0, 30.0)
+
+# Define prior distributions
+infectiousness_prior = LogNormal(log(0.3), 0.05)
+incubation_rate_prior = LogNormal(log(0.1), 0.05)
+recovery_rate_prior = LogNormal(log(0.1), 0.05)
+initial_prop_infected_prior = Beta(1, 99)
+
+# Create an instance of SIRParams
+seirparams = SEIRParams(
+    tspan = tspan,
+    infectiousness_prior = infectiousness_prior,
+    incubation_rate_prior = incubation_rate_prior,
+    recovery_rate_prior = recovery_rate_prior,
+    initial_prop_infected_prior = initial_prop_infected_prior
+)
+```
+
+# Usage Example with `ODEProcess` and predefined SIR model
+
+These
+are defined in terms of the proportions of the population in each compartment of the SEIR model
+
+
+
+# Example
+
+"""
+@kwdef struct ODEProcess{
+    P <: AbstractTuringParamModel, S, F <: Function, D <:
+                                                     Union{Dict, NamedTuple}} <:
+              EpiAwareBase.AbstractTuringEpiModel
+    "The ODE problem instance, where `P` is a subtype of `ODEProblem`."
+    params::P
+    "The solver used for the ODE problem. Default is `AutoVern7(Rodas5())`, which is an auto
+    switching solver aimed at medium/low tolerances."
+    solver::S = AutoVern7(Rodas5())
+    "A function that maps the solution object of the ODE to infection counts."
+    sol2infs::F
+    "The extra solver options for the ODE problem. Can be either a `Dict` or a `NamedTuple`
+    containing the solver options."
+    solver_options::D = Dict(:verbose => false, :saveat => 1.0)
+end
+
+@doc raw"""
+Implement the `generate_latent_infs` function for the `ODEProcess` model.
+
+Constructs a `Turing` model to generate latent infections using the specified epidemiological
+model and parameters.
+
+# Arguments
+- `epi_model::ODEProcess`: The `ODEProcess` model containing the problem definition, time steps,
+    solver, and solution-to-infections transformation function.
+- `params::ODEParams`: The initial conditions (`u0`) and parameters (`p`) for the ODE problem.
+
+# Generated quantities
+- `I_t`: The latent infections generated by solving the ODE problem with the specified parameters.
+
+# Details
+This function remakes the ODE problem with the provided initial conditions and parameters,
+    solves it using the specified solver, and then transforms the solution into latent infections
+    using the `sol2infs` function.
+
+# Example usage
+
+```julia
+using EpiAware, OrdinaryDiffEq
+r = log(2) / 7 # Growth rate corresponding to 7 day doubling time
+u0 = [1.0]
+p = [r]
+params = ODEParams(u0 = u0, p = p)
+
+# Define the ODE problem using SciML
+# We use a simple exponential growth model
+
+function expgrowth(du, u, p, t)
+    du[1] = p[1] * u[1]
+end
+prob = ODEProblem(expgrowth, u0, (0.0, 10.0), p)
```

Comment:
is it more efficient to do it this way (make a problem with fixed params and then remake) vs make it for the first time inside the call?